src/gentropy/dataset/l2g_features/other.py

```diff
@@ -0,0 +1,161 @@
+"""Methods to generate features which are not obviously categorised."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import pyspark.sql.functions as f
+
+from gentropy.common.spark_helpers import convert_from_wide_to_long
+from gentropy.dataset.gene_index import GeneIndex
+from gentropy.dataset.l2g_features.l2g_feature import L2GFeature
+from gentropy.dataset.l2g_gold_standard import L2GGoldStandard
+from gentropy.dataset.study_locus import StudyLocus
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+
+
+def common_genecount_feature_logic(
+    study_loci_to_annotate: StudyLocus | L2GGoldStandard,
+    *,
+    gene_index: GeneIndex,
+    feature_name: str,
+    genomic_window: int,
+    protein_coding_only: bool = False,
+) -> DataFrame:
+    """Computes the feature.
+
+    Args:
+        study_loci_to_annotate (StudyLocus | L2GGoldStandard): The dataset containing study loci
+            that will be used for annotation
+        gene_index (GeneIndex): Dataset containing information related to all genes in release.
+        feature_name (str): The name of the feature
+        genomic_window (int): The maximum window size to consider
+        protein_coding_only (bool): Whether to only consider protein coding genes in calculation.
+
+    Returns:
+            DataFrame: Feature dataset
+    """
+    study_loci_window = (
+        study_loci_to_annotate.df.withColumn(
+            "window_start", f.col("position") - (genomic_window / 2)
+        )
+        .withColumn("window_end", f.col("position") + (genomic_window / 2))
+        .withColumnRenamed("chromosome", "SL_chromosome")
+    )
+
+    if protein_coding_only:
+        gene_index.df = gene_index.df.filter(f.col("biotype") == "protein_coding")
```

Comment:
This changes the `df` attribute of the original gene_index.
I think there is a risk in that after you filter the df, you modify the original gene_index so that if you calculate the gene count feature the gene index won't be complete. To avoid this we usually create another object 
```suggestion
        gene_index = gene_index.filter(f.col("biotype") == "protein_coding")
```